pkg/sharings/sharings.go

```diff
@@ -0,0 +1,140 @@
+package sharings
+
+import (
+	"github.com/cozy/cozy-stack/pkg/consts"
+	"github.com/cozy/cozy-stack/pkg/couchdb"
+	"github.com/cozy/cozy-stack/pkg/permissions"
+	"github.com/cozy/cozy-stack/pkg/utils"
+	"github.com/cozy/cozy-stack/web/jsonapi"
+)
+
+// Sharing contains all the information about a sharing
+type Sharing struct {
+	SID         string `json:"_id,omitempty"`
+	SRev        string `json:"_rev,omitempty"`
+	Type        string `json:"type,omitempty"`
+	Owner       bool   `json:"owner"`
+	Desc        string `json:"desc,omitempty"`
+	SharingID   string `json:"sharing_id,omitempty"`
+	SharingType string `json:"sharing_type"`
+
+	Permissions *permissions.Set    `json:"permissions,omitempty"`
+	SRecipients []*SharingRecipient `json:"recipients,omitempty"`
+}
+
+// SharingRecipient contains the information about a recipient for a sharing
+type SharingRecipient struct {
+	Status       string
+	AccessToken  string
+	RefreshToken string
```

Comment:
It is not needed for this PR, as those fields are never modified throught json request. But it will be the case for the next steps, so it is worth nothing to add it now.